doc/tests-updates.md

```diff
@@ -0,0 +1,381 @@
+T-Conn-Id: %conn_data.conn_id%
+T-Req-Id: %conn_data.request_id%
+< ... Standard Tempesta's headers ... >
+
+```
+All headers including `T-Client-Id`, `T-Conn-Id`, `T-Req-Id` must match
+`conn_data.requests[0].expected_response`. Not all header may be matched
+exactly, especially `Date`, `Age` and other time headers. If so, headers can't
+have values smaller than expected. `T-Srv-Id` and `T-Srv-Conn-Id` headers may be
+ignored in most tests.
+
+Boolean `closed` argument is set to true if the connection was closed by the peer.
+The function is also called if the connection was reset by peer, but no response
+was received. In this case `response` is set to `None`.
+
+If received response doesn't match expected one, then an error message is generated:
+```
+Error on request-response processing!
+For request:
+---
+<Request>
+---
+Response was expected:
+---
+<Expected response>
+---
+But received response was:
+---
+<Received response>
+---
+```
+Error should also be generated if the connection was closed unexpectedly or
+connection wasn't closed as expected.
+
+
+If debug is enabled, write to file `debug_filename_%connid%` any activity on the
+connection with the timestamp: received bytes, sent bytes, and parsed messages.
+
+
+#### Mock Server
+
+Follows the same concept as mock client. Arguments:
+
+- Script. - filename with the request/response functions. The content of it
+is described later.
+
+- ID. - server id.
+
+- Interface:conns. - interfaces the client sockets will be listening and number
+of Tempesta's connections for it.
+
+- Split. - Boolean value, meaning that requests must be splited in multiple skbs.
+Don't know how to implement it yet.
+
+- Debug file. - Filename to write debug information.
+
+Server is stopped by `TERM` signal from the framework.
+
+Script contains queue of expected requests `server_data.requests[]`. If all
+requests was received, but a new request is received, then client is working in
+repeated mode, thus `conn_data.expected_request_num` must be reset to `0`.
+
+`def gen_response_func(server_data, conn_data, request)` function is used to
+give a response for the client. Returns `response` or `None`. The received
+request is compared with `server_data.requests[T-Req-Id]`
+but note, that `T-Client-Id`, `T-Conn-Id`, `T-Req-Id` values may be missed in the
+expected request.
+
+The generated response **must** contain headers copied from the request:
+`T-Client-Id`, `T-Conn-Id`, `T-Req-Id`. `T-Srv-Id` and `T-Srv-Conn-Id` must
+also be generated.
+
+If the server need to close the connection in response to request, it sets
+`conn_data.close` variable. The variable `conn_data.expect_close` can also be
+set if generated response is invalid and Tempesta will reset the connection.
+
+
+### Framework
+
+The framework now is defined not as list of actions but as HTTP flows on client
+and server. This allows to replace Tempesta by other HTTP ADC software and easily
+compare behavior between them.
```

Comment:
Each (or almost each) test case must configure an ADC and the configuration language and features are very different for different ADCs. Nginx already has it's own and pretty large test suite, but we can't just use it - mostly because of differences in configurations and features.